quil-rs/src/program/calibration.rs

```diff
@@ -33,8 +33,8 @@ use super::ProgramError;
 /// A collection of Quil calibrations (`DEFCAL` instructions) with utility methods.
 #[derive(Clone, Debug, Default, PartialEq)]
 pub struct CalibrationSet {
-    pub calibrations: Vec<Calibration>,
-    pub measure_calibrations: Vec<MeasureCalibrationDefinition>,
+    pub calibrations: HashMap<String, Calibration>,
+    pub measure_calibrations: HashMap<(String, Option<Qubit>), MeasureCalibrationDefinition>,
```

Comment:
According to [the spec](https://quil-lang.github.io/#12-6Defining-Calibrations), a calibration "key" should be a combination of name, parameters, and qubits. Essentially, it seems that the entire string after `DEFCAL` is important for disambiguating calibrations.

Reply:
Should also note that any tests using `Debug` output, e.g. with `insta`, may have unexpected test failures due to the random ordering of `HashMap`. (This can be worked around with `pretty_assertions_sorted`, but worth noting).

Possibly a better solution is to keep it as a `Vec`, but make the fields private and use methods to ensure that the items in the `Vec` are always sorted. Using `Vec::sort_by(_cached)_key` makes the most sense to me.

Reply:
Turns out existing maps and sets don't really handle the particular needs of how we want these calibration sets to work w/ a program, so I created a custom set, backed by a Vec.